Rubberduck.Deployment.Build/RubberduckPostBuildTask.cs

```diff
@@ -0,0 +1,365 @@
+        {
+            var command = $"{NetToolsDir}tlbexp.exe \"{parameters.SourceDll}\" /win32 /out:\"{parameters.SourceTlb32}\"";
+            ExecuteTask(command);
+
+            command = $"{NetToolsDir}tlbexp.exe \"{parameters.SourceDll}\" /win64 /out:\"{parameters.SourceTlb64}\"";
+            ExecuteTask(command);
+        }
+
+        private void HarvestMetadataWithWix(DllFileParameters parameters)
+        {
+            var command = $"{WixToolsDir}heat.exe file \"{parameters.SourceDll}\" -out \"{parameters.DllXml}\"";
+            ExecuteTask(command);
+
+            command = $"{WixToolsDir}heat.exe file \"{parameters.SourceTlb32}\" -out \"{parameters.TlbXml}\"";
+            ExecuteTask(command);
+        }
+
+        private IOrderedEnumerable<RegistryEntry> BuildRegistryEntriesFromMetadata(DllFileParameters parameters)
+        {
+            var builder = new RegistryEntryBuilder();
+            return builder.Parse(parameters.TlbXml, parameters.DllXml);
+        }
+
+        private void CreateInnoSetupRegistryEntries(IOrderedEnumerable<RegistryEntry> entries, DllFileParameters parameters)
+        {
+            var writer = new InnoSetupRegistryWriter();
+            var content = writer.Write(entries, parameters.DllFile, parameters.Tlb32File, parameters.Tlb64File);
+            var regFile = Path.Combine(IncludeDir, parameters.DllFile.Replace(".dll", ".reg.iss"));
+            
+            // Encoding must be UTF8 BOM
+            File.WriteAllText(regFile, content, new UTF8Encoding(true));
+        }
+
+        private bool _previousRegistrationRemoved;
+        private void RemovePreviousDebugRegistration()
+        {
+            if (_previousRegistrationRemoved)
+            {
+                return;
+            }
+
+            this.LogMessage("Removing previous debug build's registration...");
+
+            // First see if there are registry script from the previous build
+            // If so, execute them to delete previous build's keys (which may
+            // no longer exist for the current build and thus won't be overwritten)
+            var dir = Path.GetDirectoryName(RegFilePath);
+            if (Directory.Exists(dir))
+            {
+                // The current reg file is now the previous build's reg file
+                var lastRegFile = RegFilePath;
+                if (File.Exists(lastRegFile))
+                {
+                    var now = DateTime.UtcNow;
+                    if (Environment.Is64BitOperatingSystem)
+                    {
+                        var command = $"reg.exe import {lastRegFile} /reg:32";
+                        ExecuteTask(command, ignoreErrors:true);
+                        command = $"reg.exe import {lastRegFile} /reg:64";
+                        ExecuteTask(command, ignoreErrors:true);
+                    }
+                    else
+                    {
+                        var command = $"reg.exe import {lastRegFile}";
```

Comment:
reg.exe is guaranteed to be available on all windows systems we support?

Reply:
Added a comment at top of procedure